tests/e2e/lib/dpa_helpers.go

```diff
@@ -139,17 +140,32 @@ func (v *DpaCustomResource) GetNoErr() *oadpv1alpha1.DataProtectionApplication {
 }
 
 func (v *DpaCustomResource) CreateOrUpdate(spec *oadpv1alpha1.DataProtectionApplicationSpec) error {
-	cr, err := v.Get()
-	if apierrors.IsNotFound(err) {
-		v.Build(v.backupRestoreType)
-		v.CustomResource.Spec = *spec
-		return v.Create()
-	}
-	if err != nil {
+	return v.CreateOrUpdateWithRetries(spec, 3)
+}
+func (v *DpaCustomResource) CreateOrUpdateWithRetries(spec *oadpv1alpha1.DataProtectionApplicationSpec, retries int) error {
+	var (
+		err error
+		cr *oadpv1alpha1.DataProtectionApplication
+	)
+	for i := 0; i < retries; i++ {
+		if cr, err = v.Get(); apierrors.IsNotFound(err) {
+			v.Build(v.backupRestoreType)
+			v.CustomResource.Spec = *spec
+			return v.Create()
+		} else if err != nil {
+			return err
+		}
+		cr.Spec = *spec
+		if err := v.Client.Update(context.Background(), cr); err == nil {
+			return nil
+		} else if apierrors.IsConflict(err) {
+			log.Println("conflict detected during DPA CreateOrUpdate, retrying for ", retries - i , " more times")
```

Comment:
So I think this should be `retries - i - 1`  instead of `retries - i`. First time through, i is 0, so we've already done one try out of 3, with 2 left. Also, the way it's written, if we fail on the last try, we still log that we're going to retry, and we sleep 2 seconds, and then, on ending the loop, we exit since `i < retries` is false. I think to fix that the above `else if` should be `else if apierrors.IsConflict(err) && i < retries - 1`

Reply:
+1